packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -181,6 +186,7 @@ Future<vm_service.VmService> setUpVmService(
   CompileExpression? compileExpression,
   Device? device,
   GetSkSLMethod? skSLMethod,
+  GetXcodeProjectInfo? getIOSProjectInfo,
```

Comment:
This method signature is getting unwieldy `... null, null, device!.device, null, null...`
Can you swap these all to named parameters?